ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -143,12 +143,23 @@ public void onForkChoiceUpdatedResult(
         .thenAccept(
             maybeForkChoiceUpdatedResult ->
                 maybeForkChoiceUpdatedResult.ifPresent(
-                    forkChoiceUpdatedResult ->
-                        onExecutionPayloadResult(
-                            forkChoiceUpdatedResultNotification
-                                .getForkChoiceState()
-                                .getHeadBlockRoot(),
-                            forkChoiceUpdatedResult.getPayloadStatus())))
+                    forkChoiceUpdatedResult -> {
+                      if (forkChoiceUpdatedResultNotification.isTerminalBlockCall()
+                          && forkChoiceUpdatedResult.getPayloadStatus().hasInvalidStatus()) {
+                        throw new IllegalStateException(
+                            String.format(
+                                "Execution engine considers INVALID recently provided terminal block %s. "
+                                    + "Probably an execution engine misconfiguration.",
+                                forkChoiceUpdatedResultNotification
+                                    .getForkChoiceState()
+                                    .getHeadExecutionBlockHash()));
```

Comment:
I'd just log a message about this here and return rather than throwing an exception. And I'd omit the "Probably and execution engine misconfiguration." part - we can just stick to reporting what happened as we don't really know what's going on with the EL.

Reply:
LogCaptor is good, missed it, thank you